src/main/java/bobcat/executor/ExecutionUnit.java

```diff
@@ -0,0 +1,92 @@
+package bobcat.executor;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import bobcat.executor.parser.QueryParser;
+import bobcat.model.TaskList;
+import bobcat.model.task.Deadline;
+import bobcat.model.task.Event;
+import bobcat.model.task.Task;
+import bobcat.model.task.ToDo;
+
+public class ExecutionUnit {
+    private final QueryParser parser = new QueryParser();
+    private final Storage storeExecutor = new Storage();
+    private TaskList taskList = new TaskList();
+    private final String storagePath;
+
+    public ExecutionUnit(String storagePath) {
+        this.storagePath = storagePath;
+    }
+    public void initStorage() throws IOException {
+        storeExecutor.initStorage(storagePath, taskList);
+    }
+
+    public void clearStorage() {
+        taskList = new TaskList();
+    }
+
+    /**
+     * Executes the given query. ExecutionUnit will pass the query to the appropriate parser and execute the
+     * arguments according to the command recognised by the parser.
+     * @param query The given query to the chat app
+     * @return Array of strings representing information received after query is executed
+     * @throws IOException may be thrown if given storagePath to ExecutionUnit does not exist
+     */
+    public String[] executeCommand(String query) throws IOException {
```

Comment:
Perhaps you can refactor this method into multiple parts? eg. extract the logic for each action (list, bye, etc)? I would recommend to keep a method to <30 LoC